### PR TITLE
chore(script): check message during commit

### PR DESCRIPTION
**Pull request by anis-campos**
- Add script that checks message from file
- Add precommit hook to help python developers make good commit messages

### CODE REVIEW

File: check_message.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [[ -v ZSH_NAME ]]; then
```
Comment (noirbee) [inline check_message.sh:5]: How ? You asked for `bash`

File: check_message.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  setopt BASH_REMATCH
+  setopt RE_MATCH_PCRE
+  setopt KSH_ARRAYS
+fi
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+source $DIR/validator.sh
+
+MESSAGE=$(<"$1")
+echo "checking commit message: '${MESSAGE}'"
```
Comment (noirbee) [inline check_message.sh:15]: No. The error is enough

File: check_message.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+validate "$MESSAGE"
+
+echo "Message succesfully checked"
```
Comment (noirbee) [inline check_message.sh:18]: Ditto, unneeded

**Review by noirbee — CHANGES_REQUESTED** [review]
I think this makes sense only as a `commit-msg` hook, not a pre-commit one.

I doubt the hook configuration itself works as-is, given that AFAIK `pre-commit` passes files to hooks, not commit SHAs

File: check_message.sh (hunk 3 above)
Comment (anis-campos) [reply check_message.sh:18]: > I think this makes sense only as a `commit-msg` hook, not a pre-commit one.
> 
> I doubt the hook configuration itself works as-is, given that AFAIK `pre-commit` passes files to hooks, not commit SHAs

I was going to update the doc to explain you to set it up so it only runs over `commit-msg` ( basically just setting it up in `stages` in the config of `pre-commit`)

And I just figured that without COMMIT_VALIDATOR_ALLOW_TEMP, this prevents reverts and fixups... I'm adding it

File: .pre-commit-hooks.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+-   id: commit-message-validator
+    name: Commit Message Validator
+    description: Checks that commit messages are compliant with Lumapps rules.
+    entry: check_message.sh
+    language: script
```
Comment (noirbee) [inline .pre-commit-hooks.yaml:5]: Missing newline at EOF

**Review by noirbee — APPROVED** [review]

**Review by noirbee — CHANGES_REQUESTED** [review]
See comments

File: README.md [hunk 5]
```diff
@@ -242,6 +242,23 @@ jobs:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 ```
 
+## Add pre-commit plugin
```
Comment (sebastien-boulle) [inline README.md:245]: I try it on the repo itself with this branch as rev, but I wasn't able to make it work :(
Also I got some linting errors

**Review by sebastien-boulle — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (anis-campos) [reply README.md:250]: I changed the readme, maybe this is the step mising in you test : 
https://github.com/lumapps/commit-message-validator/blame/chore/precommit_hook/README.md#L266

Comment (anis-campos) [reply README.md:250]: Actually witout `--hook-type commit-msg` it can't work. I fixed the docs

**Comment by anis-campos** [comment]
https://github.com/lumapps/commit-message-validator/pull/8#discussion_r482080768

I'm finally keeping the `echo message` to prevent loss as describe inside the script.

**Comment by anis-campos** [comment]
And as discussed with @sebastien-boulle we might depreciate this in favor of a different approach:
- Warn of incorrect commit message ( `post-commit` hook)
- Prevent push with a `pre-push` hook that validates all the commit messages of the branch and prints the errors of each one

File: check_message.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+first_word=${MESSAGE%% *}
+if [[ "${first_word,,}" == merge ]]
+then
+   # ignore merge commits (merge after conflict resolution)
+  exit
+
+fi
+
+# print message so you don't loose it in case of errors
```
Comment (noirbee) [inline check_message.sh:26]: lose*

**Review by noirbee — CHANGES_REQUESTED** [review]
Not much to say otherwise

File: check_message.sh [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+# (in case you are not using `-m` option)
+printf "checking commit message:\n\n#BEGIN#\n%s\n#END#\n\n" "$(grep -v "#" <<< "$MESSAGE")"
+
+COMMIT_VALIDATOR_ALLOW_TEMP=1 COMMIT_VALIDATOR_NO_JIRA=1 validate "$MESSAGE"
```
Comment (sebastien-boulle) [inline check_message.sh:30]: Nope you don't want to deactivate JIRA validation

**Review by sebastien-boulle — APPROVED** [review]

**Review by noirbee — APPROVED** [review]

**Pull request merged**